### PR TITLE
Compact bluespace rcd

### PR DESCRIPTION
**Pull request by Kaffe-work**
### About The Pull Request

Adds a compact bluespace RCD that fits into belt slots, that is  automatically linked to the ore silo

### Why It's Good For The Game

This adds a upgraded RCD that fits into belt slots, a complaint that i and many others have about the RCD. This is done using bluespace designs, and thus it is also linked into the ore silo.

This shouldn't affect game balance too much, as it is behind several expensive techs, and is only printable at the engineering lathe.

The advantages it has is faster construction time and slightly lower construction cost, fits into belt slots, and is autolinked to the ore silo. This would allow for larger construction projects to be done without going back and forth between a lathe. 

This will not be an issue about players remote emptying the ore silo; as this can already be done swiftly at any of the protolathes, by just ejecting all materials. (At a way higher speed than it could be done using this tool)

The price on the corresponding techweb has been lowered, as well as the pre-requisites for the node itself.  

## Changelog
:cl:IradT
add: Adds a compact RCD design. Make engineering great again.
rebalance: Lowers cost of the corresponding techweb from 25000->20000
rebalance: Changed techweb slightly. 
rebalance: Allows BRCD to fit engineering belt. 
imageadd: Adds a Icon for the Bluespace RCD, Credit to Azlan
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
powercrêpe

**Comment by Qustinnus** [comment]
lol no

**Comment by TheChosenEvilOne** [comment]
You removed the plumber RCD icon, also nice icon.

**Comment by Kaffe-work** [comment]
> 
> 
> You removed the plumber RCD icon, also nice icon.

Yeah that was a mistake, thanks dream maker




> 
> 
> powercrêpe

Just like how Bag of holding is powercreep? Is it not reasonable to have more powerful versions of tools in later part of techwebs, is this not the point of science?

**Comment by Kaffe-work** [comment]
Just a question; why does mapdiffbot complain? I've not changed anything in the maps itself.

**Comment by Qustinnus** [comment]
because you have conflicts

**Comment by Kaffe-work** [comment]
> 
> 
> because you have conflicts

Yeah it was more a question what i would have to do to get it to accept the new file with both the plumbing tool and the BRCD

**Comment by Fhaxaris** [comment]
can't you already link the rcd to the ore silos with an upgrade? or am I thinking of borgs?

**Comment by CRITAWAKETS** [comment]
Ore silo upgrade is already a thing so i recommend just seperating it from this. Could add more matter storage in exchange though as the RCD's matter capacity is kinda limited.

**Comment by Kaffe-work** [comment]
> 
> 
> Ore silo upgrade is already a thing so i recommend just seperating it from this. Could add more matter storage in exchange though as the RCD's matter capacity is kinda limited.

The issue with the ore silo upgrade is that it is not automatically connected, you have to connect it using a multitool on the ore silo (in the vault). This simply merges some functionality, and makes it easier to use. I think it serves better as a separate tool than upgrades.  

There really is no balance problem here, apart from that people maybe doesn't realize that this somewhat exists. If anything, this is a quality of life upgrade.

**Comment by Ohagi-Chan** [comment]
But I like techweb nodes costing...
"Twenty-five THOUSAND points"?
> CL

**Comment by Kaffe-work** [comment]
to be fair the cost should probably be lowered to 7500 or so. 20000 is ~7 minutes of rnd time.

**Comment by HomocidlePancakes** [comment]
>My change is powercreep, but there are other powercreep items and that justifies my change.

I am speechless.

**Comment by Kaffe-work** [comment]
...there exists an RCD upgrade that lets you connect it to the silo already.

**Comment by anconfuzedrock** [comment]
Then that shouldn't exist either?

**Comment by Detective-Google** [comment]
> Nice to see the bikeshed getting a new coat of paint.

**Comment by Twaticus** [comment]
does this connect to the silo at creation? what if theres more than one silo or the original one gets destroyed? can you relink it manually afterwards? does it also take the other disk upgrades or does it already have the upgrades?

**Comment by Kaffe-work** [comment]
it connects at creation, takes first solo created. can be re-linked manually. if the linked silo gets destroyed, you need to manually re-link. it takes other upgrades

File: code/game/objects/items/RCD.dm [hunk 1]
```diff
@@ -633,6 +633,23 @@ RLD
...
 
+/obj/item/construction/rcd/bluespace
+	name = "bluespace RCD"
+	desc = "Compact, efficient bluespace RCD that is linked to the ORM,"
+	icon = 'icons/obj/BRCD.dmi'
+	icon_state = "BRCD"
+	sheetmultiplier = 4.4 //slightly better than original RCD due to tech advancement
+	w_class = WEIGHT_CLASS_SMALL
+	upgrade = TRUE
+	upgrade = RCD_UPGRADE_SILO_LINK
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/RCD.dm]: double define

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -597,11 +597,11 @@
...
 	id = "adv_rcd_upgrade"
-	display_name = "Advanced RCD designs upgrade"
-	description = "Unlocks new RCD designs."
-	design_ids = list("rcd_upgrade_silo_link")
-	prereq_ids = list("rcd_upgrade", "bluespace_travel")
-	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 25000)
+	display_name = "Bluespace-Integrated Rapid Construction"
+	description = "Bluespace RCD with automatic Silo connection."
+	design_ids = list("rcd_bluespace")
+	prereq_ids = list("adv_engi", "bluespace_travel")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/techweb/all_nodes.dm]: you didn't remove the upgrade

**Comment by ExcessiveUseOfCobblestone** [comment]
Honestly would like to just see a slightly lower techweb cost to the pre-existing upgrade, everything else seems like just a subtle way of adding buffs to said upgrade.

**Comment by Kaffe-work** [comment]
> 
> 
> Honestly would like to just see a slightly lower techweb cost to the pre-existing upgrade, everything else seems like just a subtle way of adding buffs to said upgrade.

Well, the biggest issue about the upgrade is that you have to link it to the ore silo - you need Vault access for that, something that the average engineer should not have, and shouldn't be bothered with. 

The fact that it can be placed into a toolbelt is just a quality of life upgrade. it should be a bit better if it is going to be used to justify the fact that it costs a bunch of materials and a lot of techweb cost.

**Comment by ExcessiveUseOfCobblestone** [comment]
change the upgrade to autolink then?

I'll admit the feature was probably undertuned so i don't see why we just toss in another item that fixes all those issues instead of putting it on the right one.

As for the toolbelt, that is not QoL. What can fit in what when we specifically limit slots is balance and shouldn't be ignored.

**Review by Kaffe-work — COMMENTED** [review]
> As for the toolbelt, that is not QoL. What can fit in what when we specifically limit slots is balance and shouldn't be ignored.

True, the idea here is that it is a direct upgrade at about the same point as the bluespace bag - it fits a lot of stuff. Putting a RCD in your belt should be okay at that point - this is also a design choice: cargo/eva is limited to non-bluespace RCD. 

Personally i prefer a upgraded tool than an upgrade - makes them visually distinct. Compare regular tools to the advanced ones.

File: code/modules/research/techweb/all_nodes.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/techweb/all_nodes.dm]: how do you unlock it then

**Comment by FantasticFwoosh** [comment]
Can't wait for some blood brothers to slip this inside a cake to be delivered to perma then just delete the doors. Otherwise powercrept as hell. Nice job. :+1:

**Pull request closed without merging**